### PR TITLE
fix: delete code block when selected

### LINKED ISSUE #633: Unable to delete in block-level selection after selecting code block
The code block can't be deleted by backspace if it's under block-level selection:

image

### PR DESCRIPTION
**Pull request by himself65**
Fixes: https://github.com/toeverything/blocksuite/issues/633

### CODE REVIEW

**Review by himself65 — COMMENTED** [review]
I guarantee that the current behavior has changed so I'm blocking this

**Review by doodlewind — COMMENTED** [review]
/cc @thorseraq

**Comment by himself65** [comment]
Related: https://github.com/toeverything/blocksuite/pull/539

**Comment by thorseraq** [comment]
Update: #638

**Review by lawvs — CHANGES_REQUESTED** [review]
Is it expected?

File: packages/blocks/src/page-block/default/utils.ts [hunk 1]
```diff
@@ -369,14 +355,6 @@ export function bindHotkeys(
...
 
     if (state.type === 'block') {
-      // XXX Ad-hoc for code block
-      // At the beginning of the code block,
-      // the backspace will selected the block first.
-      // The select logic already processed in the `handleLineStartBackspace` function.
-      // So we need to prevent the default delete behavior.
-      if (isDispatchFromCodeBlock(e)) {
-        return;
-      }
```
[the remark this answers is not in the document]
Comment (thorseraq) [reply packages/blocks/src/page-block/default/utils.ts:379]: Yes, this part can't be deleted

image

**Review by doodlewind — APPROVED** [review]

**Pull request merged**